From 98d6f9902dd2c086bbf0324d185b000832a97047 Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Thu, 11 May 2023 11:06:03 +0800
Subject: [PATCH 215/263] [Huawei] mm/dpool: Avoid lockdep warning with
 spin_lock_nested

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188703

--------------------------------

Use spin_lock_nested to avoid invalid lockdep warning.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 mm/dynamic_hugetlb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/dynamic_hugetlb.c b/mm/dynamic_hugetlb.c
index 303bd1fead03..1babd868fe49 100644
--- a/mm/dynamic_hugetlb.c
+++ b/mm/dynamic_hugetlb.c
@@ -40,7 +40,7 @@ static inline void dhugetlb_percpu_pool_lock_all(struct dhugetlb_pool *hpool)
 	int i;
 
 	for (i = 0; i < NR_PERCPU_POOL; i++)
-		spin_lock(&hpool->percpu_pool[i].lock);
+		spin_lock_nested(&hpool->percpu_pool[i].lock, i);
 }
 
 static inline void dhugetlb_percpu_pool_unlock_all(struct dhugetlb_pool *hpool)
-- 
2.17.1

